commit ade128a002469e7a2fa8295d4472c3eeadd4f303
Author: Stefan Kost <ensonic@users.sf.net>
Date:   Thu Feb 17 12:57:37 2011

    avidemux, flvdemux: mark delta-units in the index
    
    We need to use the 'delta' flag for delta units and not the 'none' flag.

diff --git a/gst/avi/gstavidemux.c b/gst/avi/gstavidemux.c
index a4a0525..208f9d3 100644
--- a/gst/avi/gstavidemux.c
+++ b/gst/avi/gstavidemux.c
@@ -4485,8 +4485,9 @@ gst_avi_demux_add_assoc (GstAviDemux * avi, GstAviStream * stream,
     GST_LOG_OBJECT (avi, "adding association %" GST_TIME_FORMAT "-> %"
         G_GUINT64_FORMAT, GST_TIME_ARGS (timestamp), offset);
     gst_index_add_association (avi->element_index, avi->index_id,
-        keyframe ? GST_ASSOCIATION_FLAG_KEY_UNIT : GST_ASSOCIATION_FLAG_NONE,
-        GST_FORMAT_TIME, timestamp, GST_FORMAT_BYTES, offset, NULL);
+        keyframe ? GST_ASSOCIATION_FLAG_KEY_UNIT :
+        GST_ASSOCIATION_FLAG_DELTA_UNIT, GST_FORMAT_TIME, timestamp,
+        GST_FORMAT_BYTES, offset, NULL);
     /* well, current_total determines TIME and entry DEFAULT (frame #) ... */
     gst_index_add_association (avi->element_index, stream->index_id,
         GST_ASSOCIATION_FLAG_NONE,
diff --git a/gst/flv/gstflvdemux.c b/gst/flv/gstflvdemux.c
index d976fed..86f9b14 100644
--- a/gst/flv/gstflvdemux.c
+++ b/gst/flv/gstflvdemux.c
@@ -129,8 +129,9 @@ gst_flv_demux_parse_and_add_index_entry (GstFlvDemux * demux, GstClockTime ts,
   associations[1].value = pos;
 
   gst_index_add_associationv (demux->index, demux->index_id,
-      (keyframe) ? GST_ASSOCIATION_FLAG_KEY_UNIT : GST_ASSOCIATION_FLAG_NONE,
-      2, (const GstIndexAssociation *) &associations);
+      (keyframe) ? GST_ASSOCIATION_FLAG_KEY_UNIT :
+      GST_ASSOCIATION_FLAG_DELTA_UNIT, 2,
+      (const GstIndexAssociation *) &associations);
 
   if (pos > demux->index_max_pos)
     demux->index_max_pos = pos;
